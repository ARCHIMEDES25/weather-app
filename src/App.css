.app {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.app .title {
  text-transform: uppercase;
  display: flex;
  justify-content: center;
  color: #135640;
}

.app .forecast {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
