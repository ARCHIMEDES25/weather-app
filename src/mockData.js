export const mondayList = [
  {
    dt: 1538524800,
    main: {
      temp: 7.11,
      temp_min: 6.98,
      temp_max: 7.11,
      pressure: 1009.19,
      sea_level: 1034.77,
      grnd_level: 1009.19,
      humidity: 79,
      temp_kf: 0.14,
    },
    weather: [
      {
        id: 801,
        main: 'Clouds',
        description: 'few clouds',
        icon: '02n',
      },
    ],
    clouds: {
      all: 24,
    },
    wind: {
      speed: 4.82,
      deg: 286.002,
    },
    rain: {},
    sys: {
      pod: 'n',
    },
    dt_txt: '2018-10-15 00:00:00',
  },
  {
    dt: 1538535600,
    main: {
      temp: 6.21,
      temp_min: 6.12,
      temp_max: 6.21,
      pressure: 1009.73,
      sea_level: 1035.4,
      grnd_level: 1009.73,
      humidity: 85,
      temp_kf: 0.09,
    },
    weather: [
      {
        id: 500,
        main: 'Rain',
        description: 'light rain',
        icon: '10n',
      },
    ],
    clouds: {
      all: 64,
    },
    wind: {
      speed: 2.67,
      deg: 272.501,
    },
    rain: {
      '3h': 0.03,
    },
    sys: {
      pod: 'n',
    },
    dt_txt: '2018-10-15 03:00:00',
  },
  {
    dt: 1538546400,
    main: {
      temp: 6.32,
      temp_min: 6.27,
      temp_max: 6.32,
      pressure: 1010.25,
      sea_level: 1035.84,
      grnd_level: 1010.25,
      humidity: 95,
      temp_kf: 0.05,
    },
    weather: [
      {
        id: 500,
        main: 'Rain',
        description: 'light rain',
        icon: '10n',
      },
    ],
    clouds: {
      all: 92,
    },
    wind: {
      speed: 1.71,
      deg: 233.004,
    },
    rain: {
      '3h': 0.47,
    },
    sys: {
      pod: 'n',
    },
    dt_txt: '2018-10-15 06:00:00',
  },
  {
    dt: 1538557200,
    main: {
      temp: 8.02,
      temp_min: 8.02,
      temp_max: 8.02,
      pressure: 1010.1,
      sea_level: 1035.68,
      grnd_level: 1010.1,
      humidity: 97,
      temp_kf: 0,
    },
    weather: [
      {
        id: 500,
        main: 'Rain',
        description: 'light rain',
        icon: '10d',
      },
    ],
    clouds: {
      all: 92,
    },
    wind: {
      speed: 1.86,
      deg: 171.004,
    },
    rain: {
      '3h': 0.42,
    },
    sys: {
      pod: 'd',
    },
    dt_txt: '2018-10-15 09:00:00',
  },
  {
    dt: 1538568000,
    main: {
      temp: 12.09,
      temp_min: 12.09,
      temp_max: 12.09,
      pressure: 1009.46,
      sea_level: 1034.9,
      grnd_level: 1009.46,
      humidity: 90,
      temp_kf: 0,
    },
    weather: [
      {
        id: 500,
        main: 'Rain',
        description: 'light rain',
        icon: '10d',
      },
    ],
    clouds: {
      all: 88,
    },
    wind: {
      speed: 4.77,
      deg: 246.502,
    },
    rain: {
      '3h': 0.225,
    },
    sys: {
      pod: 'd',
    },
    dt_txt: '2018-10-15 12:00:00',
  },
];

export const tuesdayList = [
  {
    dt: 1538524800,
    main: {
      temp: 7.11,
      temp_min: 5.01,
      temp_max: 7.11,
      pressure: 1009.19,
      sea_level: 1034.77,
      grnd_level: 1009.19,
      humidity: 79,
      temp_kf: 0.14,
    },
    weather: [
      {
        id: 801,
        main: 'Clouds',
        description: 'few clouds',
        icon: '02n',
      },
    ],
    clouds: {
      all: 24,
    },
    wind: {
      speed: 4.82,
      deg: 286.002,
    },
    rain: {},
    sys: {
      pod: 'n',
    },
    dt_txt: '2018-10-16 00:00:00',
  },
  {
    dt: 1538535600,
    main: {
      temp: 6.21,
      temp_min: 6.12,
      temp_max: 6.21,
      pressure: 1009.73,
      sea_level: 1035.4,
      grnd_level: 1009.73,
      humidity: 85,
      temp_kf: 0.09,
    },
    weather: [
      {
        id: 500,
        main: 'Rain',
        description: 'light rain',
        icon: '10n',
      },
    ],
    clouds: {
      all: 64,
    },
    wind: {
      speed: 2.67,
      deg: 272.501,
    },
    rain: {
      '3h': 0.03,
    },
    sys: {
      pod: 'n',
    },
    dt_txt: '2018-10-16 03:00:00',
  },
  {
    dt: 1538546400,
    main: {
      temp: 6.32,
      temp_min: 6.27,
      temp_max: 6.32,
      pressure: 1010.25,
      sea_level: 1035.84,
      grnd_level: 1010.25,
      humidity: 95,
      temp_kf: 0.05,
    },
    weather: [
      {
        id: 500,
        main: 'Rain',
        description: 'light rain',
        icon: '10n',
      },
    ],
    clouds: {
      all: 92,
    },
    wind: {
      speed: 1.71,
      deg: 233.004,
    },
    rain: {
      '3h': 0.47,
    },
    sys: {
      pod: 'n',
    },
    dt_txt: '2018-10-16 06:00:00',
  },
  {
    dt: 1538557200,
    main: {
      temp: 8.02,
      temp_min: 8.02,
      temp_max: 25.14,
      pressure: 1010.1,
      sea_level: 1035.68,
      grnd_level: 1010.1,
      humidity: 97,
      temp_kf: 0,
    },
    weather: [
      {
        id: 500,
        main: 'Rain',
        description: 'light rain',
        icon: '10d',
      },
    ],
    clouds: {
      all: 92,
    },
    wind: {
      speed: 1.86,
      deg: 171.004,
    },
    rain: {
      '3h': 0.42,
    },
    sys: {
      pod: 'd',
    },
    dt_txt: '2018-10-16 09:00:00',
  },
  {
    dt: 1538568000,
    main: {
      temp: 12.09,
      temp_min: 12.09,
      temp_max: 12.09,
      pressure: 1009.46,
      sea_level: 1034.9,
      grnd_level: 1009.46,
      humidity: 90,
      temp_kf: 0,
    },
    weather: [
      {
        id: 500,
        main: 'Rain',
        description: 'light rain',
        icon: '10d',
      },
    ],
    clouds: {
      all: 88,
    },
    wind: {
      speed: 4.77,
      deg: 246.502,
    },
    rain: {
      '3h': 0.225,
    },
    sys: {
      pod: 'd',
    },
    dt_txt: '2018-10-16 12:00:00',
  },
];

export const forecast = {
  cod: '200',
  message: 0.0032,
  cnt: 10,
  list: [...mondayList, ...tuesdayList],
  city: {
    id: 3333229,
    name: 'City of Edinburgh',
    coord: {
      lat: 55.9497,
      lon: -3.1934,
    },
    country: 'GB',
  },
};
