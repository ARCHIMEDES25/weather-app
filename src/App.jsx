import React, { Component } from 'react';
import './App.css';

import { cityToSearchFor } from './settings';
import { getForecast } from './api';
import { WeatherInfo } from './components';

class App extends Component {
  state = {
    forecast: [],
    cityName: cityToSearchFor.name,
  };

  componentDidMount = () => this.searchForecast(this.state.cityName);

  searchForecast = city =>
    getForecast(city).then(forecast => this.setState(() => ({ forecast })));

  render() {
    const { forecast, cityName } = this.state;
    return (
      <div className="app">
        <header>
          <h1 className="title">{`${cityName} forecast`}</h1>
        </header>
        <div className="forecast">
          {forecast.map(({ date, temperature }) => (
            <WeatherInfo key={date} {...{ date, temperature }} />
          ))}
        </div>
      </div>
    );
  }
}

export default App;
