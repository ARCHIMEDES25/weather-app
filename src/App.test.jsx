import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

import { forecast } from './mockData';

global.fetch = jest.fn(() => Promise.resolve({ json: () => forecast }));
global.Headers = jest.fn(() => ({ append: () => {} }));

const forecastData = [
  {
    date: '2018-10-15',
    temperature: {
      average: 7.95,
      min: 6.12,
      max: 12.09,
    },
  },
  {
    date: '2018-10-16',
    temperature: {
      average: 7.95,
      min: 5.01,
      max: 25.14,
    },
  },
];

const props = {
  state: {
    forecast: [],
    cityName: 'Edinburgh',
  },
};

describe('App component', () => {
  it('should load forecast data into state with correct shape', async done => {
    const wrapper = shallow(<App />);
    expect(wrapper.instance().state).toEqual(props.state);
    setImmediate(() => {
      expect(wrapper.instance().state.forecast).toEqual(forecastData);
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
