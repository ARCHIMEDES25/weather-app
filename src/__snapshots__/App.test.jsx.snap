// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should load forecast data into state with correct shape 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1
            className="title"
          >
            Edinburgh forecast
          </h1>
        </header>,
        <div
          className="forecast"
        >
          <WeatherInfo
            date="2018-10-15"
            temperature={
              Object {
                "average": 7.95,
                "max": 12.09,
                "min": 6.12,
              }
            }
          />
          <WeatherInfo
            date="2018-10-16"
            temperature={
              Object {
                "average": 7.95,
                "max": 25.14,
                "min": 5.01,
              }
            }
          />
        </div>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="title"
          >
            Edinburgh forecast
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edinburgh forecast",
            "className": "title",
          },
          "ref": null,
          "rendered": "Edinburgh forecast",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WeatherInfo
              date="2018-10-15"
              temperature={
                Object {
                  "average": 7.95,
                  "max": 12.09,
                  "min": 6.12,
                }
              }
            />,
            <WeatherInfo
              date="2018-10-16"
              temperature={
                Object {
                  "average": 7.95,
                  "max": 25.14,
                  "min": 5.01,
                }
              }
            />,
          ],
          "className": "forecast",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2018-10-15",
            "nodeType": "function",
            "props": Object {
              "date": "2018-10-15",
              "temperature": Object {
                "average": 7.95,
                "max": 12.09,
                "min": 6.12,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2018-10-16",
            "nodeType": "function",
            "props": Object {
              "date": "2018-10-16",
              "temperature": Object {
                "average": 7.95,
                "max": 25.14,
                "min": 5.01,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1
              className="title"
            >
              Edinburgh forecast
            </h1>
          </header>,
          <div
            className="forecast"
          >
            <WeatherInfo
              date="2018-10-15"
              temperature={
                Object {
                  "average": 7.95,
                  "max": 12.09,
                  "min": 6.12,
                }
              }
            />
            <WeatherInfo
              date="2018-10-16"
              temperature={
                Object {
                  "average": 7.95,
                  "max": 25.14,
                  "min": 5.01,
                }
              }
            />
          </div>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="title"
            >
              Edinburgh forecast
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edinburgh forecast",
              "className": "title",
            },
            "ref": null,
            "rendered": "Edinburgh forecast",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WeatherInfo
                date="2018-10-15"
                temperature={
                  Object {
                    "average": 7.95,
                    "max": 12.09,
                    "min": 6.12,
                  }
                }
              />,
              <WeatherInfo
                date="2018-10-16"
                temperature={
                  Object {
                    "average": 7.95,
                    "max": 25.14,
                    "min": 5.01,
                  }
                }
              />,
            ],
            "className": "forecast",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2018-10-15",
              "nodeType": "function",
              "props": Object {
                "date": "2018-10-15",
                "temperature": Object {
                  "average": 7.95,
                  "max": 12.09,
                  "min": 6.12,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2018-10-16",
              "nodeType": "function",
              "props": Object {
                "date": "2018-10-16",
                "temperature": Object {
                  "average": 7.95,
                  "max": 25.14,
                  "min": 5.01,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
